File: src/01-easy/retro-typewriter-art/solution.clj
(ns Solution (:gen-class))
(defn decode [tok]
  (let [ab {"sp" \space "bS" \\ "sQ" \' "nl" \newline}
        l2 (if (>= (count tok) 2) (subs tok (- (count tok) 2)) "")
        [ch num] (if (ab l2)
                   [(ab l2) (subs tok 0 (- (count tok) 2))]
                   [(last tok) (subs tok 0 (dec (count tok)))])
        n (if (empty? num) 1 (max 1 (Integer/parseInt num)))]
    (apply str (repeat n ch))))
(defn -main [& _]
  (let [line (read-line)]
    (println (apply str (map decode (clojure.string/split line #" "))))))

File: src/01-easy/retro-typewriter-art/solution.rb
STDOUT.sync = true
ab = {'sp'=>' ','bS'=>'\\','sQ'=>"'",'nl'=>"\n"}
line = gets.strip
out = ''
line.split(' ').each do |tok|
    l2 = tok[-2..]
    if ab.key?(l2)
        ch = ab[l2]; num = tok[0...-2]
    else
        ch = tok[-1]; num = tok[0...-1]
    end
    n = num.empty? ? 1 : [num.to_i, 1].max
    out += ch * n
end
puts out
